### ISSUE TITLE
Achievements does not load - blank page and "error occurred" toast [current master]

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#3036]
**Summary:** 

Every time I go into Achievements, it says "error occurred" and gives me a blank page. Logging out and in again does not fix the problem.

**System logs:**
```
06-24 19:30:56,760     [INFO ] [file-logging-thread-1] [OkHttpJsonApiClient] : Url https://tools.wmflabs.org/urbanecmbot/commonsmisc/feedback.py?user=misaochan
2019-06-24 19:30:57,817     [DEBUG] [file-logging-thread-1] [OkHttpJsonApiClient] : Response for achievements is Content-type: application/json

2019-06-24 19:30:58,947     [ERROR] [file-logging-thread-1] [AchievementsActivity] : Fetching upload count failed
java.lang.IllegalStateException: closed
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:407)
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:401)
	at okhttp3.internal.Util.bomAwareCharset(Util.java:471)
	at okhttp3.ResponseBody.string(ResponseBody.java:175)
	at fr.free.nrw.commons.mwapi.OkHttpJsonApiClient.lambda$getUploadCount$0$OkHttpJsonApiClient(OkHttpJsonApiClient.java:102)
	at fr.free.nrw.commons.mwapi.-$$Lambda$OkHttpJsonApiClient$9ZpGX0U5Zngts_jcRCFls7FtEIM.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)

2019-06-24 19:31:01,228     [INFO ] [file-logging-thread-1] [OkHttpJsonApiClient] : Url https://tools.wmflabs.org/urbanecmbot/commonsmisc/feedback.py?user=misaochan
2019-06-24 19:31:01,235     [ERROR] [file-logging-thread-1] [ContributionsFragment] : onFragmentResumed fr.free.nrw.commons.contributions.ContributionsListFragment
2019-06-24 19:31:01,286     [DEBUG] [file-logging-thread-1] [LocationServiceManager] : on location changed
2019-06-24 19:31:01,687     [DEBUG] [file-logging-thread-1] [CustomApiResult] : API response for method https://commons.wikimedia.org/w/api.php is
 <?xml version="1.0" encoding="UTF-8"?><api batchcomplete=""><query><userinfo id="4512262" name="Misaochan"/></query></api>
2019-06-24 19:31:01,691     [DEBUG] [file-logging-thread-1] [CustomMwApi] : User id is 4512262 and user name is Misaochan
2019-06-24 19:31:01,692     [DEBUG] [file-logging-thread-1] [ApacheHttpClientMediaWikiApi] : Validate login response is true
2019-06-24 19:31:02,111     [ERROR] [file-logging-thread-1] [ContributionsFragment] : Fetching upload count failed
java.lang.IllegalStateException: closed
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:407)
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:401)
	at okhttp3.internal.Util.bomAwareCharset(Util.java:471)
	at okhttp3.ResponseBody.string(ResponseBody.java:175)
	at fr.free.nrw.commons.mwapi.OkHttpJsonApiClient.lambda$getUploadCount$0$OkHttpJsonApiClient(OkHttpJsonApiClient.java:102)
	at fr.free.nrw.commons.mwapi.-$$Lambda$OkHttpJsonApiClient$9ZpGX0U5Zngts_jcRCFls7FtEIM.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)

2019-06-24 19:31:02,194     [ERROR] [file-logging-thread-1] [QuizChecker] : Fetching upload count failed
java.lang.IllegalStateException: closed
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:407)
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:401)
	at okhttp3.internal.Util.bomAwareCharset(Util.java:471)
	at okhttp3.ResponseBody.string(ResponseBody.java:175)
	at fr.free.nrw.commons.mwapi.OkHttpJsonApiClient.lambda$getUploadCount$0$OkHttpJsonApiClient(OkHttpJsonApiClient.java:102)
	at fr.free.nrw.commons.mwapi.-$$Lambda$OkHttpJsonApiClient$9ZpGX0U5Zngts_jcRCFls7FtEIM.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:764)

2019-06-24 19:31:02,225     [DEBUG] [file-logging-thread-1] [OkHttpJsonApiClient] : Response for achievements is Content-type: application/json
2019-06-24 19:31:02,277     [DEBUG] [file-logging-thread-1] [MainActivity] : Number of notifications is 0
```

**Device and Android version:** 

Samsung Galaxy S9 Android 8.1

**Commons app version:** 

Play Store alpha (current master)

**Comment by maskaravivek** [comment]
API is not returning a response for me 

https://tools.wmflabs.org/urbanecmbot/commonsmisc/feedback.py?user=Maskaravivek

@urbanecm Would you be able to check this?

**Comment by urbanecm** [comment]
Looking...

**Comment by urbanecm** [comment]
Fixed.

**Comment by maskaravivek** [comment]
@urbanecm What was the issue?

**Comment by urbanecm** [comment]
@maskaravivek thanksRecieved took 8+ minutes, and according to https://lists.wikimedia.org/pipermail/cloud/2019-June/000689.html, it can be optimalized by filtering by log_namespace. Fixed with commons-app/commonsmisc@2b5cc1ab4949d193675659aec48585ae7c39df1d.

**Comment by maskaravivek** [comment]
Works for me now. Thanks for your quick optimization @urbanecm :)
